.kuzzlerc.sample

```diff
@@ -476,19 +476,29 @@
     // (see also https://github.com/luin/ioredis)
     //
     // 1. using a single Redis database:
-    //   node:
+    //   * node:
     //     * host:
     //         The host on which Redis can be reached.
     //         Can take an IP address, an URI or a hostname
     //     * port:
     //         The port on which Redis is running its database:
-    //     * (optional) database:
+    //     * (optional) database (DEPRECATED):
     //         ID of the redis database (default: 0)
+    //         NOTE: this option is deprecated and will be removed in Kuzzle v3.
+    //               Considere using of options.db as a replacement.
+    //   * (optional) options: 
+    //       Redis specific options compatible with IORedis.
+    //       See Redis client constructor available options:
+    //       https://github.com/luin/ioredis/blob/master/API.md#redis--eventemitter
     "internalCache": {
       "backend": "redis",
+      "database": 0,
       "node": {
         "host": "localhost",
         "port": 6379
+      },
+      "options": {
+        "db": 0
```

Comment:
When dealing with deprecated options, you should make sure that behaviors are clearly defined, otherwise this makes things confusing for users and this can be detrimental when migrating Kuzzle to the next major version.

Here is what I propose:

* remove the `database` option from the `.kuzzlerc.sample` configuration file, keep it in the documentation, and explain that `options.db` takes precedence over it
* remove `options.db` from the default configuration file (`default.config.js`), and use the old `database` property instead: that way, older client configurations will still override it, and newer ones will make it ignored
* add a `@deprecated` tag in the redis service code, when using that `database` variable, to make sure we'll spot it and remove it when working on the next major version

Reply:
I'll apply your first and second point, thanks 😄